### PR TITLE
[Feature] micromamba support

### PR DESCRIPTION
**Pull request by jpfeuffer**
This enables the support for micromamba, a more lightweight, C-only, env-free alternative of conda/mamba.
It has better parallelism support through lockfiles and therefore solved https://github.com/nextflow-io/nextflow/issues/1819
for me so far.

It does not have a 100% compatible command line, therefore an if-case was needed. See https://github.com/mamba-org/mamba/issues/1654

Let me know what you think.

I was about to just use conda.executable(Name) as a config param but was worried about backwards compatibility.

### CODE REVIEW

File: modules/nf-commons/src/main/nextflow/Const.groovy [hunk 1]
```diff
@@ -58,12 +58,12 @@ class Const {
...
-    static public final long APP_TIMESTAMP = 1650736802565
+    static public final long APP_TIMESTAMP = 1651152885778
 
     /**
      * The app build number
      */
-    static public final int APP_BUILDNUM = 5697
+    static public final int APP_BUILDNUM = 5700
 
     /**
```
Comment (abhi18av) [inline modules/nf-commons/src/main/nextflow/Const.groovy:68]: This change should not be added to the PR.

File: modules/nextflow/src/main/resources/META-INF/plugins-info.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+nf-azure@0.13.1
 nf-console@1.0.3
+nf-ga4gh@1.0.3
 nf-google@1.1.4
-nf-azure@0.13.1
-nf-tower@1.4.0
-nf-ga4gh@1.0.3
\ No newline at end of file
+nf-sqldb@0.4.0
+nf-tower@1.4.0
```
Comment (abhi18av) [inline modules/nextflow/src/main/resources/META-INF/plugins-info.txt]: This should also not be a part of the PR.

**Review by abhi18av — COMMENTED** [review]
Thanks for the initiative @jpfeuffer !

I've highlighted some files which shouldn't be a part of the PR.

I was also looking into `micromamba` recently and to me it seems that it doesn't have full compatibility with `conda`.

In case of `mamba`, it _falls back_ to the `conda` layer but `micromamba` complained that "such and such wasn't implemented" in my testing earlier. Perhaps that behavior has changed recently 🤔 

Have you tested locally with a sizeable pipeline?

**Comment by jpfeuffer** [comment]
Yes, it seemed to work for me for our nf-core pipeline. Feel free to try in the copy of my gitpod workspace. Just add micromamba to the path.

https://gitpod.io#snapshot/1fd28fec-e254-4655-b148-d390435206e9

File: modules/nextflow/src/main/groovy/nextflow/conda/CondaCache.groovy [hunk 3]
```diff
@@ -252,17 +262,20 @@ class CondaCache {
...
         }
         else if( isTextFilePath(condaEnv) ) {
 
-            cmd = "${binaryName} create $opts--mkdir --yes --quiet --prefix ${Escape.path(prefixPath)} --file ${Escape.path(makeAbsolute(condaEnv))}"
+            cmd = "${binaryName} create $opts $mkdir --yes --quiet --prefix ${Escape.path(prefixPath)} --file ${Escape.path(makeAbsolute(condaEnv))}"
         }
 
         else {
-            cmd = "${binaryName} create $opts--mkdir --yes --quiet --prefix ${Escape.path(prefixPath)} $condaEnv"
+            cmd = "${binaryName} create $opts $mkdir --yes --quiet --prefix ${Escape.path(prefixPath)} $condaEnv"
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/conda/CondaCache.groovy:278]: This is not a mistake. the `opts` is expected to have a trailing space.

Comment (jpfeuffer) [reply modules/nextflow/src/main/groovy/nextflow/conda/CondaCache.groovy:278]: Yes thanks, I just saw it when I re-ran the tests. It has double spaces now.

**Review by pditommaso — CHANGES_REQUESTED** [review]
Thanks for submitting this improvement. In order to merge it, tests should be fixed, also a test should be added to validate the new option. 

Finally, a mention in the docs should be added about the new flag

**Comment by pditommaso** [comment]
This has been merged manually 383e023f. Thanks for the contribution.

**Comment by jpfeuffer** [comment]
Awesome! Thank you very much for jumping in. Last week was very busy.

**Pull request closed without merging**